RESTserver/src/main/java/com/softserve/if072/restservice/service/ForecastService.java

```diff
@@ -0,0 +1,316 @@
+            double daysDiff = (double) (currentDate.getTime() - previousDate.getTime()) / MILLISECONDS_PER_DAY;
+            productUsedSpeeds[i - 1] = usingAmounts[i] / daysDiff;
+        }
+        return productUsedSpeeds;
+    }
+
+    /**
+     * Calculate main descriptive statistics measures for using speed of product
+     *
+     * @param productStatistics - object that stores statistical information about product using
+     */
+    private void getDescriptiveStatistics(ProductStatistics productStatistics) {
+        LOGGER.trace("getDescriptiveStatistics method starts to execute.");
+        DescriptiveStatistics descriptiveStatistics = new DescriptiveStatistics(productStatistics.getProductUsingSpeeds());
+        productStatistics.setMean(descriptiveStatistics.getMean());
+        productStatistics.setMin(descriptiveStatistics.getMin());
+        productStatistics.setMax(descriptiveStatistics.getMax());
+        productStatistics.setCount(descriptiveStatistics.getN());
+        productStatistics.setRange(productStatistics.getMax() - productStatistics.getMin());
+        productStatistics.setStandardDeviation(descriptiveStatistics.getStandardDeviation());
+        productStatistics.setLowerThreeSigmaLimit(productStatistics.getMean() - 3 * productStatistics.getStandardDeviation());
+        productStatistics.setUpperThreeSigmaLimit(productStatistics.getMean() + 3 * productStatistics.getStandardDeviation());
+    }
+
+    /**
+     * Detect outliers and deal with them
+     *
+     * @param productStatistics - object that stores statistical information about product using
+     */
+    private void dealWithOutliers(ProductStatistics productStatistics) {
+        LOGGER.trace("dealWithOutliers method starts to execute.");
+        boolean hasOutliers = false;
+        double[] productUsedSpeeds = productStatistics.getProductUsingSpeeds();
+        double lowerThreeSigmaLimit = productStatistics.getLowerThreeSigmaLimit();
+        double upperThreeSigmaLimit = productStatistics.getUpperThreeSigmaLimit();
+        boolean isOutlierAtTheBeginningOrEnd = false;
+        int leftTrimBound = 0;
+        int rightTrimBound = productUsedSpeeds.length;
+
+        /*
+        * Find outliers at the beginning of the data set
+        */
+        for (int i = 0; i < productUsedSpeeds.length; i++) {
+            if (productUsedSpeeds[i] < lowerThreeSigmaLimit || productUsedSpeeds[i] > upperThreeSigmaLimit) {
+                if (i == 0) {
+                    isOutlierAtTheBeginningOrEnd = true;
+                    leftTrimBound++;
+                    continue;
+                }
+                leftTrimBound++;
+                continue;
+            }
+            LOGGER.trace("Find {} outlier(s) at the beginning of the data set.", i);
+            break;
+        }
+
+        /*
+        * Find outliers at the end of the data set
+        */
+        for (int i = productUsedSpeeds.length - 1; i >= 0; i--) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Reduce the total number of break and continue statements in this loop to use at most one. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://sonarqube.com/coding_rules#rule_key=squid%3AS135)